pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -663,6 +673,51 @@ func GetShortNameMode(ctx *types.SystemContext) (types.ShortNameMode, error) {
 	return config.shortNameMode, err
 }
 
+// AllConfiguredCredentialHelpers returns a list of all credential helpers that are configured for some registry
+// Almost all callers should use CredentialHelpersForRegistry to get configuration applicable to a specific registry.
+func AllConfiguredCredentialHelpers(sys *types.SystemContext) ([]string, error) {
+	helpers := make(map[string]bool)
+	config, err := getConfig(sys)
+	if err != nil {
+		return nil, err
+	}
+	for _, store := range config.partialV2.CredentialHelpers {
+		helpers[store] = true
+	}
+	for _, reg := range config.partialV2.Registries {
+		if reg.CredentialHelper != "" {
+			if ok := helpers[reg.CredentialHelper]; !ok {
+				helpers[reg.CredentialHelper] = true
+			}
+		}
+	}
+
+	var result []string
```

Comment:
Maps in golang are non-deterministic, so we have to move `result` up and append to it in both `for` loops. That will preserve the order _and_ make it deterministic.